Subject: [PATCH] Fixed Dialer crash issue when finish dialpad immediately

Dialer crash occurs when finish dialpad immediately.
Because dialpad fragment is hidden
although Activity has already destroyed.

diff --git a/src/com/android/dialer/DialtactsActivity.java b/src/com/android/dialer/DialtactsActivity.java
index 441501c..8cae650 100644
--- a/src/com/android/dialer/DialtactsActivity.java
+++ b/src/com/android/dialer/DialtactsActivity.java
@@ -1296,6 +1296,17 @@
     }
 
     @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        if (mSlideIn != null) {
+            mSlideIn.setAnimationListener(null);
+        }
+        if (mSlideOut != null) {
+            mSlideOut.setAnimationListener(null);
+        }
+    }
+
+    @Override
     public void onDialpadQueryChanged(String query) {
         mDialpadQuery = query;
         if (mSmartDialSearchFragment != null) {
